tools/bazel.rc

```diff
@@ -27,6 +28,7 @@ build:clang-asan --define tcmalloc=disabled
 build:clang-asan --build_tag_filters=-no_asan
 build:clang-asan --test_tag_filters=-no_asan
 build:clang-asan --define signal_trace=disabled
+build:clang-asan --copt -DADDRESS_SANITIZER=1
```

Comment:
this doesn't seem to logically belong with this PR.  Do we need it and if so what do you think about splitting it into its own?

Reply:
Ah, @htuch pointed me to the slack conversation I missed.  Yeah, let's split this out and send it his way.  We should be able to get both in today.

Reply:
@alyssawilk @htuch I just posted PR [2452](https://github.com/envoyproxy/envoy/pull/2452), which contains the google-protobuf ASAN fix in isolation. Once that change lands, I'll merge master into this PR to pick up the fix.